ecl2df/wellcompletiondata.py

```diff
@@ -63,7 +71,45 @@ def df(
         wellconnstatus_df = wellconnstatus.df(eclfiles)
         compdat_df = _merge_compdat_and_connstatus(compdat_df, wellconnstatus_df)
 
-    return _aggregate_layer_to_zone(compdat_df)
+    compdat_df = _aggregate_layer_to_zone(compdat_df)
+
+    # Add metadata as an attribute the dataframe
+    meta = _get_metadata(eclfiles)
+    # Slice meta to dataframe columns:
+    compdat_df.attrs["meta"] = {
+        column_key: meta[column_key] for column_key in compdat_df if column_key in meta
+    }
+
+    return compdat_df
+
+
+def _get_ecl_unit_system(eclfiles: EclFiles) -> EclipseUnitSystem:
+    """Returns the unit system of an eclipse deck. The options are \
+    METRIC, FIELD, LAB and PVT-M.
+
+    If none of these are found, the function returns METRIC which is the
+    default unit system in Eclipse.
+    """
+    unit_systems = [unitsystem.value for unitsystem in EclipseUnitSystem]
+    for keyword in eclfiles.get_ecldeck():
+        if keyword.name in unit_systems:
+            return EclipseUnitSystem(keyword.name)
+    return EclipseUnitSystem.METRIC
+
+
+def _get_metadata(eclfiles: EclFiles) -> Dict[str, Dict[str, Any]]:
+    """Provide metadata for the well completion data export"""
+    meta: Dict[str, Dict[str, Any]] = {}
```

Comment:
Could the `Any` be changed to
```suggestion
    meta: Dict[str, Dict[str, str]] = {}
```
or does it/should it be `Any`? If `kh_units` become an `Enum`, maybe it can be type hinted even stronger (https://stackoverflow.com/questions/52624736/type-annotations-for-enum-attribute).

Reply:
I don't think I can export an `Enum` as metadata?

I can make it `str` in this case, I guess I was thinking about other possible metadata.

Reply:
I have changed it to str